dapps/marketplace/src/pages/listings/_EarnTokensCard.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+
+import { fbt } from 'fbt-runtime'
+import { Query } from 'react-apollo'
+
+import enrollmentStatusQuery from 'queries/EnrollmentStatus'
+
+import withWallet from 'hoc/withWallet'
+import Link from 'components/Link'
+
+const EarnTokensCard = ({ wallet }) => {
+  return (
+    <Query
+      query={enrollmentStatusQuery}
+      variables={{
+        walletAddress: wallet || '0xdummyAddress'
+      }}
+      // enrollment info can change, do not cache it
+      fetchPolicy="network-only"
```

Comment:
This is very nitpicky of me. You can simplify this class and have the Link always point to `/campaigns`. This way you don't need to query `enrollmentStatus` and if the user is not signed into the Rewards program he will be automatically redirected to `/welcome`

Reply:
Thanks @sparrowDom. Have updated it.